src/Core/Services/MetadataProviders/CosmosSqlMetadataProvider.cs

```diff
@@ -291,7 +292,7 @@ private void ProcessSchema(
         private void AssertIfEntityIsAvailableInConfig(string entityName)
         {
             // If the entity is not present in the runtime config, throw an exception as we are expecting all the entities to be present in the runtime config.
-            if (!_runtimeConfig.Entities.ContainsKey(entityName))
+            if (!_runtimeConfigProvider.GetConfig().Entities.ContainsKey(entityName))
```

Comment:
Take a look at the callstack of where this function is called:
- Constructor [Contains `RuntimeConfig runtimeConfig = runtimeConfigProvider.GetConfig();`]
  - `ParseSchemaGraphQLFieldsForJoins(...)`
    - `AssertIfEntityIsAvailableInConfig(...)`

It doesn't seem correct to again call `runtimeConfigProvider.GetConfig()` in the middle of an operation when the top-level calling function has already acquired the current instance of the runtime config. 

If the runtime config changes in the middle of the operation, we should not change which runtimeconfig we are using in the current operation. The operation will need to start over again because state will be inconsistent.